src/odemis/acq/align/autofocus.py

```diff
@@ -526,12 +549,14 @@ def _CancelAutoFocus(future):
     return True
 
 
-def estimateAcquisitionTime(detector, scanner=None):
+def estimateAcquisitionTime(
+        detector: Union[model.DigitalCamera, model.Detector],
```

Comment:
DigitalCamera is a subtype of Detector. So you could replace this Union by just `model.Detector`.